### DISCUSSION TITLE
How can I access the application hostname from the server side ?

### DISCUSSION THREAD

**Discussion by RijaCloud** [nuxt/nuxt#18469] [Questions] [answered]
I need to use the domain name of my application to apply dynamic metas to it. 
Today I use window.location from the client process but it makes the meta update a bit slow because it has to wait for the whole thing to load before making the API call to the backend. 

Do you have any idea how I can get the domain name from the ssr? 
Thanks

**Comment by TheAlexLichter** [comment]
Why not passing it via `runtimeConfig` as "global variable"? ☺️

If you can't do that for whatever reason, you can use the request `Host`header to infer it.

**Reply by RijaCloud** [reply]
Hi @manniL , from where can I use the request `Host` ? 

I can't set the domain as a global variable actually because we use the same build accross multiple domain :)

**Reply by TheAlexLichter** [reply]
You can try something like:

```
<script setup>
const headers = useRequestHeaders(['host'])
const host = headers?.host ?? window.location.host
</script>
<template>
  <div>
  {{ host }}
  </div>
</template>
```

Not sure if that'll give you the right host on server-side though 🙈

**Reply by TheAlexLichter** [reply]
Alternatively you can try

`const url = useNuxtApp().ssrContext.url`

**Reply by RijaCloud** [reply]
I will try it and be back with a feedback asap :)

**Comment by ndragun92** [comment, marked as answer]
This is one of my approaches.

`/plugins/000_init.ts`
```typescript
export default defineNuxtPlugin(async (nuxtApp) => {
  console.log("Init from server/client");
  let host = "";
  // Get host in case of server side request
  if (import.meta.env.SSR) {
    host = nuxtApp?.ssrContext?.event?.req?.headers?.host || "";
  } else {
    host = window.location.host;
  }
  const origin = new URL(`https://${host}`);
  // Extract origin and set data in app context
  nuxtApp.provide("app_origin", origin.origin);
  nuxtApp.provide("app_hostname", origin.hostname);
});

```

**Reply by TheAlexLichter** [reply]
Good point, using the event via ssrContext should work too 👍🏻

**Reply by RijaCloud** [reply]
This seems to be the solution . Thanks : )
